server/sockets/join.js

```diff
@@ -61,6 +74,11 @@ module.exports = (io, socket) => {
         chats[chatId] = getChat(chatId);
       });
 
+      if(user.currentChatId <= 0) {
```

Comment:
if this is checking if user.currentChatId is null, then just do it this way

also what's the logic behind this, sounds like you're avoiding the user from opening a chat if they're already blocked, if that's so it's not needed as if a user blocks another we can just perform almost the same action for when a user closes a chat

```suggestion
      if(!user.currentChatId) {
```

Reply:
> if this is checking if user.currentChatId is null, then just do it this way
> 
> also what's the logic behind this, sounds like you're avoiding the user from opening a chat if they're already blocked, if that's so it's not needed as if a user blocks another we can just perform almost the same action for when a user closes a chat

I added this because there was a bug preventing pairing between anonymous users and users logged in with email. I believe I understand the root cause and would like to discuss it further on Discord 

maybe its a local bug, if you want i can just remove them ?

Reply:
Please avoid dming me and discuss it here for future references